internal/trace/trace.go

```diff
@@ -16,35 +16,74 @@ package trace
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"os"
+	"sort"
+	"strings"
 
 	"go.opencensus.io/trace"
-	"golang.org/x/xerrors"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	ottrace "go.opentelemetry.io/otel/trace"
 	"google.golang.org/api/googleapi"
 	"google.golang.org/genproto/googleapis/rpc/code"
 	"google.golang.org/grpc/status"
 )
 
+const (
+	telemetryPlatformTracing              = "GOOGLE_API_GO_EXPERIMENTAL_TELEMETRY_PLATFORM_TRACING"
+	telemetryPlatformTracingOpenCensus    = "opencensus"
+	telemetryPlatformTracingOpenTelemetry = "opentelemetry"
+)
+
+var (
+	// TODO(chrisdsmith): What is the correct name for the OpenTelemetry tracer?
+	OpenTelemetryTracerName string = "cloud.google.com/go/internal/trace"
+)
+
+func IsOpenCensusTracingEnabled() bool {
+	return !IsOpenTelemetryTracingEnabled()
+}
+
+func IsOpenTelemetryTracingEnabled() bool {
```

Comment:
Can we reduce this to a singleton check somehow?  As written this means we're going to be interrogating the environment variables on every span start/end, but something like a sync.once or similar could do the env check once and retain the results of the lookup.

Reply:
Yes, great point. Not sure how I missed this, really appreciate the comment.

Reply:
Updated, PTAL.

Reply:
Thanks for making this change!